Makefile

```diff
@@ -1,320 +1,37 @@
-CLEAN_TARGETS =
-
-CLEAN_TARGETS += clean-artifacts
-.PHONY: clean-artifacts
-# Removes the ./bin directory.
-clean-artifacts:
-	-rm -rf $(INSTALL_PREFIX)
-
-CLEAN_TARGETS += clean-object-files
-.PHONY: clean-object-files
-# Runs go clean to remove any executables or other object files.
-clean-object-files:
-	go clean ./...
-
-CLEAN_TARGETS += clean-vendor
-.PHONY: clean-vendor
-# Removes the ./vendor directory.
-clean-vendor:
-	-rm -rf $(VENDOR_DIR)
-
-CLEAN_TARGETS += clean-tmp
-.PHONY: clean-tmp
-# Removes the ./vendor directory.
-clean-tmp:
-	-rm -rf $(TMP_PATH)
-
-# Keep this "clean" target here after all `clean-*` sub tasks
-.PHONY: clean
-## Cleans the project, removes all generated code/bins and vendor packages
-clean: $(CLEAN_TARGETS)
-
-# -------------------------------------------------------------------
-# build the binary executable (to ship in prod)
-# -------------------------------------------------------------------
+include ./make/help.mk
+include ./make/out.mk
+include ./make/find-tools.mk
+include ./make/go.mk
+include ./make/git.mk
+include ./make/dev.mk
+include ./make/format.mk
+include ./make/lint.mk
+include ./make/deploy.mk
+include ./make/test.mk
+include ./make/docker.mk
 
 .PHONY: build
 ## Build the operator
-build: prebuild-check deps check-go-format
-	@echo "building $(BINARY_SERVER_BIN)..."
-	operator-sdk generate k8s
-	go build -v -o $(BINARY_SERVER_BIN) cmd/manager/main.go
-
-# -------------------------------------------------------------------
-# deploy
-# -------------------------------------------------------------------
-
-# to watch all namespaces, keep namespace empty
-APP_NAMESPACE ?= ""
-LOCAL_TEST_NAMESPACE ?= "local-test"
-.PHONY: local
-## Run Operator locally
-local: deploy-rbac build deploy-crd
-	operator-sdk up local --namespace=$(APP_NAMESPACE)
+build: ./out/operator
 
-.PHONY: deploy-rbac
-## Setup service account and deploy RBAC
-deploy-rbac:
-	@-oc login -u system:admin
-	@-oc create -f deploy/service_account.yaml
-	@-oc create -f deploy/role.yaml
-	@-oc create -f deploy/role_binding.yaml
-
-.PHONY: deploy-crd
-## Deploy CRD
-deploy-crd:
-	@-oc apply -f deploy/crds/devopsconsole_v1alpha1_component_crd.yaml
-	@-oc apply -f deploy/crds/devopsconsole_v1alpha1_gitsource_crd.yaml
-
-.PHONY: deploy-operator
-## Deploy Operator
-deploy-operator: deploy-crd
-	oc create -f deploy/operator.yaml
-
-.PHONY: deploy-clean
-## Deploy a CR as test
-deploy-clean:
-	@-oc delete project $(LOCAL_TEST_NAMESPACE)
-
-.PHONY: deploy-test
-## Deploy a CR as test
-deploy-test:
-	@-oc new-project $(LOCAL_TEST_NAMESPACE)
-	@-oc apply -f examples/devopsconsole_v1alpha1_component_cr.yaml
+.PHONY: clean
+clean:
+	$(Q)-rm -rf ${V_FLAG} ./out
+	$(Q)-rm -rf ${V_FLAG} ./vendor
+	$(Q)go clean ${X_FLAG} ./...
+
+./vendor: Gopkg.toml Gopkg.lock
+	$(Q)dep ensure ${V_FLAG} -vendor-only
+
+./out/operator: ./vendor $(shell find . -path ./vendor -prune -o -name '*.go' -print)
+	#$(Q)operator-sdk generate k8s
```

Comment:
Fine to have the operator-sdk as a built time dependency since we are not doing a `generate k8s`.